docs/recipes/reducers/06-NormalizingStateShape.md

```diff
@@ -0,0 +1,208 @@
+				comment : ".....",
+			},
+			{
+				id : "comment5",
+				author : {username : "user3", name : "User 3"},
+				comment : ".....",
+			}
+        ]    
+    }
+    // and repeat many times
+]
+```
+
+Notice that the structure of the data is a bit complex, and some of the data is repeated.  This is a concern for several reasons:
+
+- When a piece of data is duplicated in several places, it becomes harder to make sure that it is updated appropriately.
+- Nested data means that the corresponding reducer logic has to be more nested or more complex.  In particular, trying to update a deeply nested field can become very ugly very fast.
+- Since immutable data updates require all ancestors in the state tree to be copied and updated as well, and new object references will cause connected UI components to re-render, an update to a deeply nested data object could force totally unrelated UI components to re-render even if the data they're displaying hasn't actually changed.
+
+Because of this, the recommended approach to managing relational or nested data in a Redux store is to treat a portion of your store as if it were a database, and keep that data in a _normalized_ form.  
+
+## Designing a Normalized State
+
+The basic concepts of normalizing data are:
+
+- Each type of data gets its own "table" in the state.
+- Each "data table" should store the individual items in an object, with the IDs of the items as keys and the items themselves as the values.
+- Any references to individual items should be done by storing the item's ID.
+- Arrays of IDs should be used to indicate ordering.
+
+An example of a normalized state structure for the blog example above might look like:
+
+```js
+{
+    posts : {
+        byId : {
+            "post1" : {
+                id : "post1",
+				author : "user1",
+				body : "......",
+				comments : ["comment1", "comment2"]    
+            },
+            "post2" : {
+				id : "post2",
+				author : "user2",
+				body : "......",
+				comments : ["comment3", "comment4", "comment5"]    
+            }
+        }
+        allIds : ["post1", "post2"]
+    },
+    comments : {
+        byId : {
+            "comment1" : {
+                id : "comment1",
+                author : "user2",
+                comment : ".....",
+            },
+            "comment2" : {
+                id : "comment2",
+                author : "user3",
+                comment : ".....",
+            },
+            "comment3" : {
+                id : "comment3",
+                author : "user3",
+                comment : ".....",
+            },
+            "comment4" : {
+                id : "comment4",
+                author : "user1",
+                comment : ".....",
+            },
+            "comment5" : {
+                id : "comment5",
+                author : "user3",
+                comment : ".....",
+            },
+        },
+        allIds : ["comment1", "comment2", "comment3", "commment4", "comment5"]
```

Comment:
Yes - the idea is that for _every_ entity type, you'd have both the lookup table and the list of all items.